bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/HueBridgeHandler.java

```diff
@@ -591,56 +627,97 @@ private void handleExceptionWhileCreatingUser(Exception ex) {
+        final String lightId = lightStatusListener.getLightId();
+        if (lightStatusListeners.put(lightId, lightStatusListener) != null) {
+            return false;
+        }
+
+        if (hueBridge != null) {
             startLightPolling();
-            // inform the listener initially about all lights and their states
-            for (FullLight light : lastLightStates.values()) {
+            final FullLight light = lastLightStates.get(lightId);
+            if (light != null) {
                 lightStatusListener.onLightAdded(hueBridge, light);
             }
         }
-        return result;
+
+        return true;
     }
 
     @Override
     public boolean unregisterLightStatusListener(LightStatusListener lightStatusListener) {
-        boolean result = lightStatusListeners.remove(lightStatusListener);
-        if (result) {
-            // stop stop light polling
-            if (lightStatusListeners.isEmpty()) {
-                stopLightPolling();
-            }
+        final String lightId = lightStatusListener.getLightId();
+        if (lightStatusListeners.remove(lightId) == null) {
+            return false;
+        }
+
+        if (discoveryListener == null && lightStatusListeners.isEmpty()) {
+            stopLightPolling();
         }
-        return result;
+
+        return true;
     }
 
     @Override
     public boolean registerSensorStatusListener(SensorStatusListener sensorStatusListener) {
-        boolean result = sensorStatusListeners.add(sensorStatusListener);
-        if (result && hueBridge != null) {
-            // start sensor polling only if a sensor handler has been registered
+        final String sensorId = sensorStatusListener.getSensorId();
+        if (sensorStatusListeners.put(sensorId, sensorStatusListener) != null) {
+            return false;
+        }
+
+        if (hueBridge != null) {
             startSensorPolling();
-            // inform the listener initially about all sensors and their states
-            for (FullSensor sensor : lastSensorStates.values()) {
+            final FullSensor sensor = lastSensorStates.get(sensorId);
+            if (sensor != null) {
                 sensorStatusListener.onSensorAdded(hueBridge, sensor);
             }
         }
-        return result;
+
+        return true;
     }
 
     @Override
     public boolean unregisterSensorStatusListener(SensorStatusListener sensorStatusListener) {
-        boolean result = sensorStatusListeners.remove(sensorStatusListener);
-        if (result) {
-            // stop sensor polling
-            if (sensorStatusListeners.isEmpty()) {
-                stopSensorPolling();
-            }
+        final String sensorId = sensorStatusListener.getSensorId();
+        if (sensorStatusListeners.remove(sensorId) == null) {
+            return false;
+        }
+
+        if (discoveryListener == null && lightStatusListeners.isEmpty()) {
+            stopLightPolling();
```

Comment:
stopSensorPolling

Reply:
Moved start and stop polling. So not necessary anymore.